Compile error when compiling rustc for ios.

This is d0d37075a54b68df99c46d42655a06c575eecd3f. 

I am trying to follow [this tutorial](https://www.bignerdranch.com/blog/building-an-ios-app-in-rust-part-1/) to learn something about iOS. It contains a step about cross compiling rustc for iOS:

~~~
$ mkdir build && cd build
$ ../configure --target=armv7-apple-ios,armv7s-apple-ios,i386-apple-ios,aarch64-apple-ios,x86_64-apple-ios --prefix=/opt/rust-ios-toolchains

$ make -j8
...tons of output...

rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/librustc_bitflags
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7-apple-ios/lib/liballoc
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7s-apple-ios/lib/liballoc
rustc: x86_64-apple-darwin/stage2/lib/rustlib/aarch64-apple-ios/lib/liballoc
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-ios/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/liballoc
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7-apple-ios/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7s-apple-ios/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/i386-apple-ios/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/aarch64-apple-ios/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcollections
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-ios/lib/libstd
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7-apple-ios/lib/libstd
warning: dropping unsupported crate type `dylib` for target `x86_64-apple-ios`
warning: dropping unsupported crate type `dylib` for target `armv7-apple-ios`
rustc: x86_64-apple-darwin/stage2/lib/rustlib/armv7s-apple-ios/lib/libstd
rustc: x86_64-apple-darwin/stage2/lib/rustlib/i386-apple-ios/lib/libstd
warning: dropping unsupported crate type `dylib` for target `armv7s-apple-ios`
warning: dropping unsupported crate type `dylib` for target `i386-apple-ios`
rustc: x86_64-apple-darwin/stage2/lib/rustlib/aarch64-apple-ios/lib/libstd
warning: dropping unsupported crate type `dylib` for target `aarch64-apple-ios`
rustc: x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libstd
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 error: use of unstable library feature 'result_fold': has not seen enough usage to justify its position in the standard library
../src/libstd/sys/unix/backtrace.rs:133     result::fold(iter, (), |_, _| ())
                                            ^~~~~~~~~~~~
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 help: add #![feature(result_fold)] to the crate attributes to enable
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 error: use of deprecated item: has not seen enough usage to justify its position in the standard library, #[deny(deprecated)] on by default
../src/libstd/sys/unix/backtrace.rs:133     result::fold(iter, (), |_, _| ())
                                            ^~~~~~~~~~~~
error: aborting due to 2 previous errors
make: *** [x86_64-apple-darwin/stage2/lib/rustlib/armv7-apple-ios/lib/stamp.std] Error 101
make: *** Waiting for unfinished jobs....
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 error: use of unstable library feature 'result_fold': has not seen enough usage to justify its position in the standard library
../src/libstd/sys/unix/backtrace.rs:133     result::fold(iter, (), |_, _| ())
                                            ^~~~~~~~~~~~
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 help: add #![feature(result_fold)] to the crate attributes to enable
../src/libstd/sys/unix/backtrace.rs:133:5: 133:17 error: use of deprecated item: has not seen enough usage to justify its position in the standard library, #[deny(deprecated)] on by default
../src/libstd/sys/unix/backtrace.rs:133     result::fold(iter, (), |_, _| ())
                                            ^~~~~~~~~~~~
error: aborting due to 2 previous errors
make: *** [x86_64-apple-darwin/stage2/lib/rustlib/armv7s-apple-ios/lib/stamp.std] Error 101
~~~

Seems like an honest bug - I wonder why I only see this with these config flags? Am I doing it wrong?

--- a/src/libstd/sys/unix/backtrace.rs
+++ b/src/libstd/sys/unix/backtrace.rs
@@ -107,8 +107,6 @@ use sys_common::backtrace::*;
 #[cfg(all(target_os = "ios", target_arch = "arm"))]
 #[inline(never)]
 pub fn write(w: &mut Write) -> io::Result<()> {
-    use result;
-
     extern {
         fn backtrace(buf: *mut *mut libc::c_void,
                      sz: libc::c_int) -> libc::c_int;
@@ -127,10 +125,10 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     let cnt = unsafe { backtrace(buf.as_mut_ptr(), SIZE as libc::c_int) as usize};
 
     // skipping the first one as it is write itself
-    let iter = (1..cnt).map(|i| {
-        print(w, i as isize, buf[i], buf[i])
-    });
-    result::fold(iter, (), |_, _| ())
+    for i in 1..cnt {
+        try!(print(w, i as isize, buf[i], buf[i]))
+    }
+    Ok(())
 }
 
 #[cfg(not(all(target_os = "ios", target_arch = "arm")))]

